src/ai/defaultai.cc

```diff
@@ -2978,14 +2976,21 @@ bool DefaultAI::construct_building(const Time& gametime) {
 					number_of_same_nearby += same_it->second;
 				}
 
-				// Considering distance to a warehouse NOCOM
-				if (!bo.requires_supporters && !bo.inputs.empty()) {
-					prio -= wh_distance_malus;
-				}
-				// some "independent" space consumers can be pushed to bigger distance from wh
-				if (bo.is(BuildingAttribute::kSpaceConsumer) && bo.inputs.empty() &&
-				    !bo.is(BuildingAttribute::kSupportingProducer)) {
-					prio += wh_distance_malus;  // push them farer from nearest wh
+				// Considering distance to wh for productionsites:
+				// All productionsites in one place
+
+				if (bo.is(BuildingAttribute::kWell) && bo.is(BuildingAttribute::kRanger) &&
+				    bo.is(BuildingAttribute::kHunter)) {
+					prio += wh_distance_malus;
+				} else if (bo.is(
+				              BuildingAttribute::
+				                 kSpaceConsumer)) {  // ??? &&
+					                                  // bo.is(BuildingAttribute::kSupportingProducer))
+					                                  // {
+					prio += wh_distance_malus;
+				} else if (!bo.inputs.empty() && !bo.ware_outputs.empty() &&
+				           !bo.is(BuildingAttribute::kSupportingProducer)) {
```

Comment:
I think Supporting producers should not be considered here but we should ensure that the malus is not applied to buildings requiring suppporters